tests/integration/states/file.py

```diff
@@ -437,44 +499,52 @@ def test_managed_contents(self):
         managed_files = {}
         state_keys = {}
         for typ in ('bool', 'str', 'int', 'float', 'list', 'dict'):
-            managed_files[typ] = tempfile.mkstemp()[1]
-            state_keys[typ] = 'file_|-{0} file_|-{1}_|-managed'.format(typ, managed_files[typ])
-        try:
-            with salt.utils.fopen(state_file, 'w') as fp_:
-                fp_.write(textwrap.dedent('''\
-                bool file:
-                  file.managed:
-                    - name: {bool}
-                    - contents: True
-
-                str file:
-                  file.managed:
-                    - name: {str}
-                    - contents: Salt was here.
-
-                int file:
-                  file.managed:
-                    - name: {int}
-                    - contents: 340282366920938463463374607431768211456
-
-                float file:
-                  file.managed:
-                    - name: {float}
-                    - contents: 1.7518e-45  # gravitational coupling constant
+            fd_, managed_files[typ] = tempfile.mkstemp()
 
-                list file:
-                  file.managed:
-                    - name: {list}
-                    - contents: [1, 1, 2, 3, 5, 8, 13]
+            # Release the handle so they can be removed in Windows
```

Comment:
@terminalmage You have to close them here otherwise when state is applied later on salt can't manage the files because they are considered 'in-use' by windows and the tests fail.